ink-wrapper-types/src/calls.rs

```diff
@@ -115,21 +118,27 @@ impl ExecCall {
 /// Reperesents a contract call to a payable method that still needs the value transferred to be specified.
 /// Use the `with_value()` method to set the value.
 #[derive(Debug, Clone)]
-pub struct ExecCallNeedsValue {
+pub struct ExecCallNeedsValue<T: scale::Decode + Send> {
     /// The account id of the contract to call.
     pub account_id: AccountId,
     /// The encoded data of the call.
     pub data: Vec<u8>,
+    /// A marker for the type to decode the result into.
+    _return_type: PhantomData<T>,
 }
 
-impl ExecCallNeedsValue {
+impl<T: scale::Decode + Send> ExecCallNeedsValue<T> {
```

Comment:
Rant: I'm still unsure about this, but I guess it makes sense for drink. But it feels like we won't be able to generalize this over other backends, because they won't return the value to us. Perhaps we can worry when we're adding another backend.

Reply:
I thought in such cases we'd have to return `()`. It used to work when there was `aleph_client` backend still in the repo.

Reply:
Sure, let's not waste time on it now, and just revisit when we're adding subxt.